megatron/core/inference/batch_dimensions_utils.py

```diff
@@ -0,0 +1,306 @@
+                    int(cuda_graph_step_size) / CUDAGraphBatchDimensionBuilder.CUDA_GRAPH_ROUNDER
+                )
+            )
+            # Make sure divisible by TP size
+            cuda_graph_step_size = math.ceil(cuda_graph_step_size / tp_size) * tp_size
+
+            # Cuda graph token counts.
+            if num_cuda_graphs == 1:
+                cuda_graph_token_counts = [cuda_graph_max_tokens]
+            else:
+                cuda_graph_token_counts = list(
+                    range(cuda_graph_step_size, cuda_graph_max_tokens, cuda_graph_step_size)
+                )
+                if cuda_graph_token_counts[-1] != cuda_graph_max_tokens:
+                    cuda_graph_token_counts.append(cuda_graph_max_tokens)
+                cuda_graph_token_counts.reverse()
+
+        cuda_graph_batch_dimensions_list = []
+        if num_cuda_graphs is None:
+            cuda_graph_batch_dimensions_list = []
+        elif (
+            not cuda_graph_mixed_prefill_count
+            or cuda_graph_mixed_prefill_count <= 0
+            or not use_cuda_graphs_for_non_decode_steps
+        ):  # decode only
+            for size in cuda_graph_token_counts:
+                add_if_valid(
+                    token_count=min(size, max_requests),
+                    prefill_req_count=0,
+                    decode_req_count=min(size, max_requests),
+                )
+        else:
+            for size in cuda_graph_token_counts:
+                add_if_valid(
+                    token_count=min(size, max_requests),
+                    prefill_req_count=0,
+                    decode_req_count=min(size, max_requests),
+                )
+                add_if_valid(
+                    token_count=size,
+                    prefill_req_count=min(cuda_graph_mixed_prefill_count, max_requests),
+                    decode_req_count=min(size, max_requests)
+                    - min(cuda_graph_mixed_prefill_count, max_requests),
+                )
+                # We need to ensure the prefill requests are shorter than the max sequence length,
+                # considering the one decode token is used for prefill request construction
+                prefill_only_minimal_num = max(
+                    cuda_graph_mixed_prefill_count,
+                    math.ceil(size / max(1, max_sequence_length - 1)),
+                )
+                if prefill_only_minimal_num < max_requests:
+                    add_if_valid(
+                        token_count=size,
+                        prefill_req_count=max(prefill_only_minimal_num, min(max_requests, size)),
+                        decode_req_count=0,
+                    )
+
+        # Remove duplicates and sort by prefill token count
+        cuda_graph_batch_dimensions_list = list(set(cuda_graph_batch_dimensions_list))
+        cuda_graph_batch_dimensions_list.sort(
+            key=lambda x: ((x.token_count - x.decode_req_count), x.decode_req_count), reverse=True
+        )
```

Comment:
This is just a style thing; up to you.

Since this seems to be the only way you'll be sorting these objects, you can directly define
```
def __lt__(self, other: "CudaGraphBatchDimensions") -> bool:
    self_val = (self.token_count  - self.decode_req_count,  self.decode_req_count)
    other_val = (other.token_count - other.decode_req_count, other.decode_req_count)
    return self_val < other_val
```
and not have to pass in `lambda` whenever you sort.

Reply:
But perhaps more importantly, I don't see any reason to pre-sort this list - unless I'm missing something about the CG building step.

If we're just going to call `min` or `max` on this list anyway, and that's `O(n)` compared to the `O(n lg n)` of sorting it, why even sort it in the first place?